src/sources/util/unix.rs

```diff
@@ -0,0 +1,19 @@
+use std::os::unix::fs::PermissionsExt;
+use std::{fs, fs::remove_file, path::Path};
+
+use crate::internal_events::UnixSocketFileDeleteError;
+
+pub fn change_socket_permissions(path: &Path, perms: Option<u32>) -> crate::Result<()> {
+    if let Some(mode) = perms {
+        match fs::set_permissions(path, fs::Permissions::from_mode(mode)) {
+            Ok(_) => info!(message = "Socket permissions updated.", permission = mode),
```

Comment:
I wonder if this shouldn't be a `debug!` instead. It's not something I think we need to expose at the `info` level.

Reply:
Agree, changed to debug!